src/sidebar/oauth-auth.js

```diff
@@ -79,14 +90,23 @@ function auth($http, settings) {
 
   // Set a timeout to refresh the access token a few minutes before it expires.
   function refreshAccessTokenBeforeItExpires(tokenInfo) {
-    var delay = tokenInfo.expiresIn * 1000;
+    // The delay, in milliseconds, before we will poll again to see if it's
+    // time to refresh the access token.
+    var delay = 5000;
```

Comment:
One problem with making it longer is that it won't work very well in dev environments where it's handy to set the lifetime to 10s. In production though I think 30s would be fine and I guess you could always change the delay in dev (and not commit the change)